### DISCUSSION TITLE
HDAO v HBAO

### DISCUSSION THREAD

**Discussion by exostalker** [OGSR/OGSR-Engine#288] [Q&A] [answered]
First of all, thank you for the non-stop amazing work.

When choosing HBAO in the settings menu (at least on DX11), the very high option is never saved. The option always reverts back to high after game restart. However, no issues with this if HDAO is chosen.

It's interesting how much the tesselation/bump is affected by SSAO. Is there any truth to HBAO=nvidia and HDAO=ati? I prefer HDAO in most cases, but in some it seems way overpronounced - here's HDAO (very high):
![ss_v_08-31-20_23-58-38_(l01_escape)](https://user-images.githubusercontent.com/18196930/91770174-61dcf780-ebe9-11ea-8ef1-c5c6c352e9af.jpg)
HBAO (high since very high cannot be set, always reverts):
![ss_v_08-31-20_23-59-57_(l01_escape)](https://user-images.githubusercontent.com/18196930/91770248-85a03d80-ebe9-11ea-8b7d-e45fd23c3bee.jpg)

Also, it seems that the bump/tesselation disappears quite abruptly when you're getting close to the surfaces in shadow. Not sure if this can even be fixed, but just throwing it out there. This is a bit distracting, for example with Sid's bunker doors.
![ss_v_08-31-20_23-58-52_(l01_escape)](https://user-images.githubusercontent.com/18196930/91770266-8df87880-ebe9-11ea-8c26-e646e1d8cb21.jpg)

A few more examples of the HUGE difference in bumps/tesselation that SSAO makes:
HDAO:
![ss_v_09-01-20_00-01-57_(l01_escape)](https://user-images.githubusercontent.com/18196930/91770430-d44dd780-ebe9-11ea-8e12-b84dd7862965.jpg)

HBAO:
![ss_v_09-01-20_00-00-45_(l01_escape)](https://user-images.githubusercontent.com/18196930/91770412-cd26c980-ebe9-11ea-9407-fec0250fb94b.jpg)

Anyways, most likely most of this is just a comment, as it probably works as intended due to these 2 being different technologies? I have an nvidia card, it's interesting how hbao is causing visual glitches with some grass nearby the player being darkened slightly (not visible in screenshots, but distracting when walking), so will stick with hdao for now.

**Comment by xrSimpAI** [comment, marked as answer]
> When choosing HBAO in the settings menu (at least on DX11), the very high option is never saved. The option always reverts back to high after game restart. However, no issues with this if HDAO is chosen.

Apparently, this was done on purpose. Why - I don't know, I can assume that these are crutches from some bugs with the SSAO.

> Is there any truth to HBAO=nvidia and HDAO=ati?

It seems like yes. At least I've read about it too.

> I prefer HDAO in most cases, but in some it seems way overpronounced - here's HDAO (very high)

Maybe it depends on specific textures / or their settings / bumps.

P.S. The stairs in the first and second screenshots hurt my eyes. Please enable SMAA :grin:

**Comment by exostalker** [comment]
Thanks for the response :)

After testing all of the possible options out, I'm pretty happy with both default SSAO and HDAO. It's interesting how HBAO, which is meant for nvidia presumably, looks so awful on grass on my nvidia card. It's very distracting, but both of these other two work fine.

After checking shader files, I found that the normal map intensity is controlled in ssao_hdao_new.ps in this line:
static float g_fHDAONormalScale     = 0.10f;	// Scaling factor to control the effect the normals have 

Lowering this value makes them less and less intense, so I'll play around with the values, pretty sure I'll get in a sweetspot where it's not as pronounced but also makes a difference.

However, even after playing around with other values in the file, I couldn't find any that control the distance to which SSAO is disabled. Is there any chance you know what needs to be changed for this? I'll illustrate my example with two videos, where the bumps that SSAO incresases disappear near the player, very distracting so would be lovely to figure out if it can be tweaked to at least minimise it:

SSAO default (high):
https://www.youtube.com/watch?v=VVWmGMo5WgQ

HDAO very high:
https://www.youtube.com/watch?v=0ljcjqCuLUs
